dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java

```diff
@@ -414,8 +416,9 @@ public static ResultSummary generateResultSummary(VerificationTask task, ProverE
             summary.append(result).append("\n");
         } else {
             // Litmus-specific output format that matches with Herd7 (as good as it can)
-            if (p.getFilterSpecification() != null) {
-                summary.append("Filter ").append(p.getFilterSpecification()).append("\n");
+            Expression filterSpec = p.getFilterSpecification();
+            if (filterSpec != null && (!(filterSpec instanceof BoolLiteral bLit) || !bLit.getValue())) {
```

Comment:
You can remove the null check. I changed filterSpec to always have a value, but it seems I forgot to update this code.